cdap-kubernetes/src/main/java/co/cask/cdap/k8s/runtime/KubeTwillController.java

```diff
@@ -0,0 +1,240 @@
+  }
+
+  @Override
+  public Future<Set<String>> restartInstances(Map<String, ? extends Set<Integer>> runnableToInstanceIds) {
+    throw new UnsupportedOperationException("restartInstances is currently not supported");
+  }
+
+  @Override
+  public Future<String> restartInstances(String runnable, int instanceId, int... moreInstanceIds) {
+    throw new UnsupportedOperationException("restartInstances is currently not supported");
+  }
+
+  @Override
+  public Future<String> restartInstances(String runnable, Set<Integer> instanceIds) {
+    throw new UnsupportedOperationException("restartInstances is currently not supported");
+  }
+
+  @Override
+  public Future<Map<String, LogEntry.Level>> updateLogLevels(Map<String, LogEntry.Level> logLevels) {
+    CompletableFuture<Map<String, LogEntry.Level>> f = new CompletableFuture<>();
+    f.completeExceptionally(new UnsupportedOperationException("updateLogLevels is currently not supported"));
+    return f;
+  }
+
+  @Override
+  public Future<Map<String, LogEntry.Level>> updateLogLevels(String runnableName,
+                                                             Map<String, LogEntry.Level> logLevelsForRunnable) {
+    CompletableFuture<Map<String, LogEntry.Level>> f = new CompletableFuture<>();
+    f.completeExceptionally(new UnsupportedOperationException("updateLogLevels is currently not supported"));
+    return f;
+  }
+
+  @Override
+  public Future<String[]> resetLogLevels(String... loggerNames) {
+    CompletableFuture<String[]> f = new CompletableFuture<>();
+    f.completeExceptionally(new UnsupportedOperationException("resetLogLevels is currently not supported"));
+    return f;
+  }
+
+  @Override
+  public Future<String[]> resetRunnableLogLevels(String runnableName, String... loggerNames) {
+    CompletableFuture<String[]> f = new CompletableFuture<>();
+    f.completeExceptionally(new UnsupportedOperationException("resetLogLevels is currently not supported"));
+    return f;
+  }
+
+  @Override
+  public RunId getRunId() {
+    return runId;
+  }
+
+  @Override
+  public Future<Command> sendCommand(Command command) {
+    CompletableFuture<Command> f = new CompletableFuture<>();
+    f.completeExceptionally(new UnsupportedOperationException("sendCommand is currently not supported"));
+    return f;
+  }
+
+  @Override
+  public Future<Command> sendCommand(String runnableName, Command command) {
+    CompletableFuture<Command> f = new CompletableFuture<>();
+    f.completeExceptionally(new UnsupportedOperationException("sendCommand is currently not supported"));
+    return f;
+  }
+
+  @Override
+  public Future<? extends ServiceController> terminate() {
+    if (!completion.isDone()) {
+      // delete the deployment, then delete the config map
+      try {
+        V1DeleteOptions deleteOptions = new V1DeleteOptions();
+        AppsV1Api appsApi = new AppsV1Api(apiClient);
+        appsApi.deleteNamespacedDeployment(name, kubeNamespace, deleteOptions, null, null, null, null);
```

Comment:
Better implement this method using the `async` variant of the delete method and complete the future from the callback (actually the first callback will trigger the async deletion of config map, which triggers the completion of termination future).